pulp_gem/app/tasks/publishing.py

```diff
@@ -48,32 +84,45 @@ def publish(repository_version_pk):
-                pk__in=publication.repository_version.content
-            ).order_by("-pulp_created"):
-                for content_artifact in content.contentartifact_set.all():
-                    published_artifact = PublishedArtifact(
-                        relative_path=content_artifact.relative_path,
-                        publication=publication,
-                        content_artifact=content_artifact,
-                    )
-                    published_artifact.save()
-                if re.fullmatch(r"[0-9.]*", content.version):
-                    specs.append(Key(content.name, content.version))
-                    old_ver = latest_versions.get(content.name)
-                    if old_ver is None or version.parse(old_ver) < version.parse(content.version):
-                        latest_versions[content.name] = content.version
-                else:
-                    prerelease_specs.append(Key(content.name, content.version))
-            latest_specs = [Key(name, version) for name, version in latest_versions.items()]
-
-            _publish_specs(specs, "specs.4.8", publication)
-            _publish_specs(latest_specs, "latest_specs.4.8", publication)
-            _publish_specs(prerelease_specs, "prerelease_specs.4.8", publication)
+    with GemPublication.create(repository_version, pass_through=True) as publication:
+        specs = []
+        latest_versions = {}
+        prerelease_specs = []
+        gems = []
+        gemspecs = []
+        for content in (
+            GemContent.objects.filter(pk__in=publication.repository_version.content)
+            .only("name", "version")
+            .order_by("-pulp_created")
+            .iterator()
+        ):
+            if re.fullmatch(r"[0-9.]*", content.version):
+                specs.append(Key(content.name, content.version))
+                old_ver = latest_versions.get(content.name)
+                if old_ver is None or version.parse(old_ver) < version.parse(content.version):
+                    latest_versions[content.name] = content.version
+            else:
+                prerelease_specs.append(Key(content.name, content.version))
+            gems.append(content.relative_path)
+            gemspecs.append(content.gemspec_path)
+        latest_specs = [Key(name, ver) for name, ver in latest_versions.items()]
+
+        _publish_specs(specs, "specs.4.8", publication)
+        _publish_specs(latest_specs, "latest_specs.4.8", publication)
+        _publish_specs(prerelease_specs, "prerelease_specs.4.8", publication)
+        _create_index(
+            publication,
+            path="",
+            links=[
+                "gems/",
+                "quick/Marshal.4.8/",
+                "specs.4.8",
+                "latest_specs.4.8",
+                "prerelease_specs.4.8",
+            ],
+        )
+        _create_index(publication, path="gems/", links=gems)
+        _create_index(publication, path="quick/", links=[])
```

Comment:
Would this not need a link to "quick/Marshal.4.8/"?

Reply:
You are correct.